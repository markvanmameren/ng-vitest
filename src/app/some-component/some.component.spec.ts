import { provideExperimentalZonelessChangeDetection } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { beforeEach, describe, expect, it } from 'vitest';
import { SomeComponent } from './some.component';

describe('SomeComponent', () => {
  let fixture: ComponentFixture<SomeComponent>;

  beforeEach(async () => {
    TestBed.configureTestingModule({
      imports: [SomeComponent],
      providers: [provideExperimentalZonelessChangeDetection()],
    }).compileComponents();

    fixture = TestBed.createComponent(SomeComponent);
  });

  it('should have an h1', () => {
    const h1: HTMLHeadingElement | null =
      fixture.nativeElement.querySelector('h1');
    const headerText = h1?.innerHTML;

    expect(headerText).toBe('Some header');
  });

  it('should match snapshot', () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
