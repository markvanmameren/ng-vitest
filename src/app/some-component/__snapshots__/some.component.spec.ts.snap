// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SomeComponent > should create the app 1`] = `
<div
  _nghost-a-c145517223=""
  id="root0"
  ng-version="18.2.9"
>
  <h1
    _ngcontent-a-c145517223=""
  >
    Some header
  </h1>
  <p
    _ngcontent-a-c145517223=""
  >
    Some content
  </p>
</div>
`;
